### PR TITLE
polish swap review sheet

### PR DESCRIPTION
**Pull request by greg-schrammel**
Fixes BX-####
Figma link (if any):

## What changed (plus any additional context for devs)

- better format route part percentage
- fix fallback icon when protocol has no icon
- fixes a bug where if the route changed while you are in the review sheet sometimes the "routing via" row value vanished

### CODE REVIEW

**Review by derHowie — APPROVED** [review]
💅

File: src/entries/popup/components/ExternalImage/ExternalImage.tsx [hunk 1]
```diff
@@ -137,9 +137,9 @@ const ExternalImage = (props: ExternalImageProps) => {
           ? { borderRadius: props.borderRadius }
           : {}),
         boxShadow: isLoading || error ? undefined : props.boxShadow,
-        height: props.height,
         overflow: error ? 'visible' : 'clip',
-        width: props.width,
+        height: props.height && (Number(props.height) ?? props.height),
+        width: props.width && (Number(props.width) ?? props.width),
```
Comment (estebanmino) [inline src/entries/popup/components/ExternalImage/ExternalImage.tsx:142]: i'm a bit confused, why is this?

**Review by estebanmino — APPROVED** [review]

File: src/entries/popup/components/ExternalImage/ExternalImage.tsx (hunk 1 above)
Comment (greg-schrammel) [reply src/entries/popup/components/ExternalImage/ExternalImage.tsx:142]: when passing `height="20"` it didn't work but `height={20}` did, I think it expect to have a unit like `height="20px"` when it's a string (?)

**Review by DanielSinclair — APPROVED** [review]
Looks good to me!

**Pull request merged**